test/one/nio/serial/JsonReaderTest.java

```diff
@@ -108,10 +108,56 @@ public void nullableFields() throws IOException, ClassNotFoundException {
         Assert.assertTrue(x.set instanceof Set && x.set.isEmpty());
     }
 
+    @Test
+    public void unicodeEscaping() throws IOException, ClassNotFoundException {
+        String s = "{\"string2\":\"A\u003D\u003dZ\"}";
+        JsonReader reader = new JsonReader(s.getBytes());
+        Custom x = reader.readObject(Custom.class);
+        Assert.assertEquals("A==Z", x.string2);
+    }
+
+    @Test
+    public void shouldReadBooleansAndNumbersForStringField() throws IOException, ClassNotFoundException {
+        String s = "{\"string2\": null}";
+        JsonReader reader = new JsonReader(s.getBytes());
+        Custom x = reader.readObject(Custom.class);
+        Assert.assertNull(x.string2);
+
+        s = "{\"string2\": false}";
+        reader = new JsonReader(s.getBytes());
+        x = reader.readObject(Custom.class);
+        Assert.assertEquals("false", x.string2);
+
+        s = "{\"string2\": true}";
+        reader = new JsonReader(s.getBytes());
+        x = reader.readObject(Custom.class);
+        Assert.assertEquals("true", x.string2);
+
+        s = "{\"string2\": 123}";
+        reader = new JsonReader(s.getBytes());
+        x = reader.readObject(Custom.class);
+        Assert.assertEquals("123", x.string2);
+    }
+
+    @Test
+    public void shouldReadStringForBooleanAndNumberField() throws IOException, ClassNotFoundException {
+        String s = "{\"aBoolean\": \"true\"}";
+        JsonReader reader = new JsonReader(s.getBytes());
+        Custom x = reader.readObject(Custom.class);
+        Assert.assertTrue(x.aBoolean);
+
+        s = "{\"longValue\": \"123\"}";
+        reader = new JsonReader(s.getBytes());
+        x = reader.readObject(Custom.class);
+        Assert.assertEquals(123L, (long)x.longValue);
```

Comment:
`x.longValue.longValue()` might look better than casting.

Reply:
The cast looks nicer to me.